massa-ledger-worker/src/ledger_db.rs

```diff
@@ -585,21 +633,22 @@ impl LedgerDB {
         &self,
         addr: &Address,
     ) -> std::collections::BTreeMap<Vec<u8>, Vec<u8>> {
+        let serialized_key = datastore_prefix_from_address(addr);
         let handle = self.db.cf_handle(LEDGER_CF).expect(CF_ERROR);
 
         let mut opt = ReadOptions::default();
-        opt.set_iterate_upper_bound(end_prefix(data_prefix!(addr)).unwrap());
+        opt.set_iterate_upper_bound(end_prefix(&serialized_key).unwrap());
 
         self.db
             .iterator_cf_opt(
                 handle,
                 opt,
-                IteratorMode::From(data_prefix!(addr), Direction::Forward),
+                IteratorMode::From(&serialized_key, Direction::Forward),
             )
             .flatten()
             .map(|(key, data)| {
                 (
-                    key.split_at(ADDRESS_SIZE_BYTES + 1).1.to_vec(),
+                    key.split_at(ADDRESS_SIZE_BYTES + 2).1.to_vec(), // DIRTY FIX: It's supposed to be 1 but 2 works. I couldn't find why.
```

Comment:
I think this should be investigated, maybe check with @Eitu33

Reply:
As @Eitu33 is absent, I will take over on this. As the key was previously considered as byte array we had the length of the full key at start of the serialized form and we deserialized everything in one batch.
Now that the key is a more specified type we don't need the length to be encoded as we know how much byte when to deserialize and how we interpret them. We only need a length for datastore key as it's the only one that has a variable size. 
This function return the datastore key stripped and so now that we only need it for datastore variant the byte is just before the datastore key.

Something that we need to change is that this byte is useful when serialising to send it through the network in messages/ledger part but not in the database as there is no data after the key. So we need to have 2 different serializers/deserializers